app/styles/ui/toolbar/_toolbar.scss

```diff
@@ -33,19 +33,20 @@
 
   .toolbar-button {
     &.push-pull-button {
-      width: 230px;
+      width: 100%;
     }
   }
 
   .toolbar-button {
     &.branch-toolbar-button {
-      width: 230px;
+      width: 100%;
     }
     &.revert-progress {
       width: 230px;
     }
     &.toolbar-dropdown-arrow-button {
       width: 39px;
+      flex-shrink: 0;
```

Comment:
Prevent shrinking the dropdown arrow button. Keep them always pinned to the right.